prac_08/convert_miles_km.py

```diff
@@ -0,0 +1,40 @@
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.properties import StringProperty
+
+MILES_TO_KMS = 1.60934
+
+
+class ConvertMilesKm(App):
+    """ ConvertMilesKm is an app for converting miles to kilometres """
+    result = StringProperty()
+
+    def build(self):
+        """ build the Kivy app from the kv file """
+        self.title = "Convert Miles to KMs"
+        self.root = Builder.load_file('convert_miles_km.kv')
+        self.result = "0.0"
+        return self.root
+
+    def handle_conversion(self, value):
+        """ Handle conversion, output result to self.result """
+        try:
+            result = float(value) * MILES_TO_KMS
+            self.result = str(result)
+        except ValueError:
+            self.result = str(0.0)
+
+    def handle_increment(self, increment):
+        """ handle up/down button press, update the text input with new value, call conversion function """
+        try:
```

Comment:
Another option here, to avoid converting from str to int twice, would be to assign the int() or 0 value (as an integer) to the number_of_miles local variable in the try/except, and then just increment the latter in the following line.